src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -92,7 +102,19 @@ const onEnableDedicatedGatewayChange = (
   if (dedicatedGatewayOriginallyEnabled === newValue) {
     currentValues.set("sku", baselineValues.get("sku"));
     currentValues.set("instances", baselineValues.get("instances"));
-    currentValues.set("costPerHour", baselineValues.get("costPerHour"));
+
+    if (newValue === true) {
```

Comment:
there is already a 'if newValue === true' block. Move this there

Reply:
the path inside this if statement will not hit the other if newValue===true block, it returns at the end of the if statement.
I can reorganize the code to only check if newValue===true once, but I wasn't sure if that was acceptable, so just made minimal change to existing code.

Reply:
dedicatedGatewayOriginallyEnabled === newValue ====> set everything to the baseline. it is already computed on init. Set costPerHour to baseline as well. 

This newValue ===true logic is irrespective of whether dedicatedGatewayOriginallyEnabled === newValue or not. So move it to that section